backend/src/main/java/asegroup1/api/models/landregistry/LandRegistryData.java

```diff
@@ -0,0 +1,333 @@
+	}
+
+	public void setStreetName(String streetName) {
+		addAddrConstraint(Selectable.street, streetName);
+	}
+
+	public void setTownName(String townName) {
+		addAddrConstraint(Selectable.town, townName);
+	}
+
+	public void setLocality(String locality) {
+		addAddrConstraint(Selectable.locality, locality);
+	}
+
+	public void setDistrict(String district) {
+		addAddrConstraint(Selectable.district, district);
+	}
+
+	public void setCounty(String county) {
+		addAddrConstraint(Selectable.county, county);
+	}
+
+	public void setPostCode(String postCode) throws InvalidParameterException {
+		if (postCode.charAt(postCode.length() - 4) != 32) {
+			throw new InvalidParameterException("Post Code: must contain a space in the correct position");
+		} else {
+			addAddrConstraint(Selectable.postcode, postCode);
+		}
+	}
+
+	private void addTransConstraint(Selectable selectable, String name, String value, boolean isString) {
+		constraints.put(selectable, new TransConstraint(name, value, isString));
+	}
+
+	private void addTransConstraint(Selectable selectable, String name, String value) {
+		addTransConstraint(selectable, name, value, false);
+	}
+
+
+	public void setPropertyType(PropertyType propertyType) {
+		addTransConstraint(Selectable.propertyType, "propertyType/skos:prefLabel", parseEnumAsString("lrcommon", propertyType.toString()));
+	}
+
+	public void setNewBuild(Boolean newBuild) {
+		addTransConstraint(Selectable.newBuild, "newBuild", newBuild.toString());
+	}
+
+	public void setEstateType(EstateType estateType) {
+		addTransConstraint(Selectable.estateType, "estateType/skos:prefLabel", parseEnumAsString("lrcommon", estateType.toString()));
+	}
+
+	public void setTransactionCategory(TransactionCategory transactionCategory) {
+		addTransConstraint(Selectable.transactionCategory, "transactionCategory/skos:prefLabel", parseEnumAsString("lrppi", transactionCategory.toString()));
+	}
+
+	public void setPricePaid(int pricePaid) {
+		addTransConstraint(Selectable.pricePaid, "pricePaid", pricePaid + "");
+	}
+
+	public void setTransactionDate(LocalDate date) {
+		addTransConstraint(Selectable.transactionDate, "transactionDate", date.toString(), true);
+	}
+
+	public boolean hasConstraint(Selectable selectable) {
+		return constraints.containsKey(selectable);
+	}
+
+	public String getConstraint(Selectable selectable) {
+		if (hasConstraint(selectable)) {
+			return constraints.get(selectable).getvalue();
+		} else {
+			return null;
+		}
+	}
+
+	public boolean removeConstraint(Selectable selectable) {
+		if (hasConstraint(selectable)) {
+			constraints.remove(selectable);
+			return true;
+		} else {
+			return false;
+		}
+
+	}
+
+	private String parseEnumAsString(String namespace, String enumStr) {
+		return namespace + ":" + enumStr.replace('_', '-');
+	}
+
+	public boolean setConstraint(String name, String value) {
+		Selectable selectable;
+		if (value.length() == 0) {
+			return true;
+		}
+		if (name.length() < 2) {
+			return false;
```

Comment:
maybe throw an exception saying why it wasn't changed rather than just returning false?